go/multipooler/manager/manager.go

```diff
@@ -1108,13 +1070,302 @@ func (pm *MultiPoolerManager) ConfigureSynchronousReplication(ctx context.Contex
 	return nil
 }
 
+// applyAddOperation adds new standbys to the standby list (idempotent)
+func applyAddOperation(currentStandbys []*clustermetadatapb.ID, newStandbys []*clustermetadatapb.ID) []*clustermetadatapb.ID {
+	updatedStandbys := append([]*clustermetadatapb.ID{}, currentStandbys...)
+	existingMap := make(map[string]bool)
+	for _, standby := range currentStandbys {
+		existingMap[generateApplicationName(standby)] = true
+	}
+	for _, newStandby := range newStandbys {
+		if !existingMap[generateApplicationName(newStandby)] {
+			updatedStandbys = append(updatedStandbys, newStandby)
+		}
+	}
+	return updatedStandbys
+}
+
+// applyRemoveOperation removes standbys from the standby list (idempotent)
+func applyRemoveOperation(currentStandbys []*clustermetadatapb.ID, standbysToRemove []*clustermetadatapb.ID) []*clustermetadatapb.ID {
+	removeMap := make(map[string]bool)
+	for _, standby := range standbysToRemove {
+		removeMap[generateApplicationName(standby)] = true
+	}
+	var updatedStandbys []*clustermetadatapb.ID
+	for _, standby := range currentStandbys {
+		if !removeMap[generateApplicationName(standby)] {
+			updatedStandbys = append(updatedStandbys, standby)
+		}
+	}
+	return updatedStandbys
+}
+
+// applyReplaceOperation replaces the entire standby list
+func applyReplaceOperation(newStandbys []*clustermetadatapb.ID) []*clustermetadatapb.ID {
+	return newStandbys
+}
+
+// UpdateSynchronousStandbyList updates PostgreSQL synchronous_standby_names by adding,
+// removing, or replacing members. It is idempotent and only valid when synchronous
+// replication is already configured.
+func (pm *MultiPoolerManager) UpdateSynchronousStandbyList(ctx context.Context, operation multipoolermanagerdata.StandbyUpdateOperation, standbyIDs []*clustermetadatapb.ID, reloadConfig bool, consensusTerm int64, force bool) error {
+	if err := pm.checkReady(); err != nil {
+		return err
+	}
+
+	if err := pm.lock(ctx); err != nil {
+		return err
+	}
+	defer pm.unlock()
+
+	pm.logger.Info("UpdateSynchronousStandbyList called",
+		"operation", operation,
+		"standby_ids", standbyIDs,
+		"reload_config", reloadConfig,
+		"consensus_term", consensusTerm,
+		"force", force)
+
+	// Validate and update consensus term following consensus rules
+	if err := pm.validateAndUpdateTerm(ctx, consensusTerm, force); err != nil {
```

Comment:
@deepthi - I think we need this here as well. Similar to SetPrimaryConnInfo. Tagging you to double check my reasoning here.

Reply:
We'll need to refactor this when we implement consensus properly, but when we get there, the term should already have been updated before this RPC is called. This RPC should not be the one that is updating the term.